src/Component/ColorPicker/index.js

```diff
@@ -10,56 +10,54 @@ import PrimaryButton from 'Component/Button/Primary';
 import SecondaryButton from 'Component/Button/Secondary';
 import SimpleInput from 'Component/Input/Simple';
 import Toggle from 'Component/Toggle';
+import {formatColorAsRgbOrHex, getHexColorOrNull, getRgbaColorOrNull, isRgb, rgbFormat} from 'Util/colorHexRgbHelper';
 
 import './ColorPicker.scss';
 
 export default function ColorPicker(props) {
     const [rgbColor, setRgbColor] = useState('');
+    const [colorInput, setColorInput] = useState('');
     const [isRgbFormat, setIsRgbFormat] = useState(false);
+    const inputError = useRef(false);
 
     useEffect(() => {
         if (!props.color) return;
 
-        setFormattedRgbColor(props.color);
-        setIsRgbFormat(
-            isRgb(props.color)
-        );
+        const rgb = isRgb(props.color);
+        setRgbColor(rgbFormat(props.color));
+        setColorInput(formatColorAsRgbOrHex(props.color, rgb))
+        setIsRgbFormat(rgb);
     }, [props.color]);
 
-    function setFormattedRgbColor(color) {
-        setRgbColor(
-            rgbFormat(color)
-        );
+    function updateRgbToggle(rgbModeOn) {
+        setIsRgbFormat(rgbModeOn);
+        setColorInput(formatColorAsRgbOrHex(rgbColor, rgbModeOn));
     }
 
-    function isHex(color) {
-        return color.startsWith('#');
+    function updateSelectedColor(color) {
+        setColorInput(formatColor(color));
+        setRgbColor(rgbFormat(color));
     }
 
-    function hexFormat(color) {
-        return isHex(color)
-            ? color
-            : colord(color).toHex()
+    function updateColorInput(color) {
+        const validColor = isRgbFormat
```

Comment:
CONSDER: Encapsulate this into `isValidColor(color, isRgbFormat)`. One import less as a result because you don't use `getRgbaColorOrNull` and `getHexColorOrNull` anywhere else.